Sources/SwiftGenKit/Parsers/Parser.swift

```diff
@@ -23,9 +25,23 @@ public protocol Parser {
 }
 
 public extension Parser {
-  func parse(paths: [Path]) throws {
+  func searchAndParse(paths: [Path], filter: Filter) throws {
     for path in paths {
-      try parse(path: path)
+      try searchAndParse(path: path, filter: filter)
+    }
+  }
+
+  func searchAndParse(path: Path, filter: Filter) throws {
```

Comment:
The fact that those two methods are not declared in the protocol too means that it will use static dispatch and not dynamic dispatch. I think that there's something with confusing situations arising in such cases, like If one day we have a parser that is a subclass of another parser, it might take the implementation of the superclass (if called on a variable declared as the superclass at compile-time) and not the child class (real type of that variable at runtime)…

Not sure this will really happen, and not sure I remember that thing and behaviour properly, but both something to keep in mind and something to double-check

Reply:
Hmmm, the `searchAndParse` will be static, but `parse(path:` is defined in the protocol, so that should be dispatched dynamically (I think). If not, that'd seem like a swift bug to me? 😕

Reply:
Yeah, but I think there is (that bug) indeed. Anyway, worth checking. And tbh, this doesn't hurt much to put them in the protocol anyway, making those implementations in extensions "default implementations of the protocol" (instead of being seen as "additional static methods added to all conforming types" for cases like here when it's not declared as a requirement on the protocol)

Reply:
Ok this seems that there's no bug after all, false alarm… I think?

```swift
protocol Parser {
//  func parse(path: String)
}

extension Parser {
  func parse(path: String) {
    print("Parser parsing \(path)")
  }
  func searchAndParse(path: String) {
    print("Search >>>")
    parse(path: path)
    print("<<< Search")
  }
}

class Parent: Parser {
  func parse(path: String) {
    print("Parent parsing \(path)")
  }
}

class Child: Parent {
  override func parse(path: String) {
    print("Child parsing \(path)")
  }
}

let parser: Parent = Child()
parser.parse(path: "path1")
parser.searchAndParse(path: "path2")
```

Output if I DO declare "func parse(path: String)" in the protocol (uncomment it):
```
Child parsing path1
Search >>>
Child parsing path2
<<< Search
```

Output if I DON'T declare "func parse(path: String)" in the protocol (keep it commented):
```
/*
Child parsing path1
Search >>>
Parser parsing path2
<<< Search
*/
```
So in this case, `parseAndSearch` calls the implementation on `Parser`… which calls the `parse` defined on `Parser`, so despite the `parser` variable being a `Child()` and declared as `: Parent`, it calls neither the `parse()` on `Child` nor the one on `Parent`, but the one on the protocol…

But as long as `parse(path: )` is declared as a requirement on the `protocol`, at least this one will be dynamic it seems, so… 😅:+1: ?